File: VehicleServiceCenterNew/BackEnd/src/main/java/lk/vsc/service/SupplierService.java
package lk.vsc.service;

import lk.vsc.DTO.UpdateJobPrice;
import lk.vsc.entity.Supplier;

import java.util.List;

public interface SupplierService {

    Supplier addSupplier(Supplier supplier);
    void deleteSupplier(int supplier);
    Supplier updateSupplier(Supplier supplier);
    Supplier searchBySupplierName(String searchSupplierName);
    List<Supplier> getAllSupplier();
    List<Supplier> getSupplierNames(String supplierCompany);
    UpdateJobPrice searchServiceDetailsByNumber(String serviceId);
    String updateSupplierPayments(UpdateJobPrice u);
}

File: VehicleServiceCenterNew/FrontEnd/src/app/Service/supplier.service.ts
import { Injectable } from '@angular/core';
import {Supplier} from "../Model/Supplier";
import {HttpClient} from "@angular/common/http";
import {environment} from "../../environments/environment";

const URL = '/SupplierController';

@Injectable({
  providedIn: 'root'
})
export class SupplierService {


  public constructor(private http: HttpClient) { }

  addSupplier(supplier: Supplier) {

    return this.http.post<Supplier>(environment.backend_url + URL + '/addSupplier', supplier);

  }

  UpdateSupplierDetails(updateSupplierDetails: Supplier) {


    // return this.http.put<Supplier>(environment.backend_url + URL + '/updateSupplier/'+updateSupplierDetails.nic,updateSupplierDetails);
    return this.http.post<Supplier>(environment.backend_url + URL + '/updateSupplier',updateSupplierDetails);

  }


  deleteSupplier(supplierId: number) {

    return this.http.delete<string>(environment.backend_url + URL + '/deleteSupplier/' + supplierId);

  }

  searchSupplierDetails(searchSupplierName: string) {
    return this.http.get<Supplier>(environment.backend_url + URL + '/searchBySupplierName/'+searchSupplierName);
  }


  getAllSupplier(){

    return this.http.get<Array<Supplier>>(environment.backend_url + URL + '/getAllSupplier');

  }

  getSuppliersNames(companyName:string) {

    return this.http.get<Array<Supplier>>(environment.backend_url + URL + '/getSuppliersNames/'+companyName);

  }

  updateSupplier(updateSupplierVar: Supplier) {
    return this.http.post<Supplier>(environment.backend_url + URL + '/updateSupplier',updateSupplierVar);
  }
}
